src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -386,15 +386,20 @@ public async Task SaveChangesAsync(IList<UnspentOutputs> unspentOutputs, IEnumer
 
                         FetchCoinsResponse result = await this.inner.FetchCoinsAsync(new[] { unspent.TransactionId }).ConfigureAwait(false);
 
+                        foreach (var utxo in result.UnspentOutputs)
+                        {
+                            this.logger.LogTrace("UTXO fetched from disk: {0}", utxo == null ? "null" : utxo.ToString());
```

Comment:
utxo.ToString will cause us to perform ToString of every utxo even when we have trace disabled.
I think LogTrace should alredy perform .tostring internally

Reply:
well I see it happens on many places in this commit, so won't repeat but this applies on every .ToString I see in this commits

Reply:
it happens only during SaveAsync/Rewind but we can save anyway cpu here

Reply:
Yeah no need to do `ToString()` just set the param (it will happen inside the logger anyway

Reply:
Oh yes of course @MithrilMan! I will fix now :)